### PR TITLE
Add uncompressed KTX files to baked texture output

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10295/HighFidelity-Beta-PR13283-b420c20.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10295/HighFidelity-Beta-PR13283-b420c20.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10295/HighFidelity-Beta-Interface-PR13283-b420c20.exe

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10308/HighFidelity-Beta-PR13283-b420c20.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10308/HighFidelity-Beta-PR13283-b420c20.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10308/HighFidelity-Beta-Interface-PR13283-b420c20.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10308/HighFidelity-Beta-PR13283-b420c20.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10308/HighFidelity-Beta-Interface-PR13283-b420c20.exe

**Comment by Qliemillar** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10421/HighFidelity-Beta-PR13283-b420c20.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10421/HighFidelity-Beta-PR13283-b420c20.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10421/HighFidelity-Beta-Interface-PR13283-b420c20.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10421/HighFidelity-Beta-PR13283-b420c20.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10421/HighFidelity-Beta-Interface-PR13283-b420c20.exe

**Comment by qligmaloney** [comment]
FAIL @conklin94122 Android Interface crashes on launch. Defect does not occur on Master (Build 8363). Test plan passed on Desktop Interface with no other new defect discovered.
Log  attached

[PR13293 Android hifi-log.txt](https://github.com/highfidelity/hifi/files/2081874/PR13293.Android.hifi-log.txt)

**Comment by huffman** [comment]
@qligmaloney Did this happen more than once?

**Comment by huffman** [comment]
@qligmaloney Testing locally, I see it too. I'll keep you posted.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10471/HighFidelity-Beta-PR13283-24182d3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10471/HighFidelity-Beta-PR13283-24182d3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10471/HighFidelity-Beta-PR13283-24182d3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10471/HighFidelity-Beta-Interface-PR13283-24182d3.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10474/HighFidelity-Beta-PR13283-3e65e0b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10474/HighFidelity-Beta-PR13283-3e65e0b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10474/HighFidelity-Beta-Interface-PR13283-3e65e0b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10474/HighFidelity-Beta-PR13283-3e65e0b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10474/HighFidelity-Beta-Interface-PR13283-3e65e0b.exe

**Comment by qlipdahlin** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
@conklin94122 PASS

Both devices loaded the interface and the sandbox. Imported script and desktop saw 4 bottom boxes with success on them while mobile saw 2 top boxes with success on them. Only issues that occurred were ones known to occur on master.

File: libraries/baking/src/TextureBaker.cpp [hunk 1]
```diff
@@ -124,47 +126,51 @@ void TextureBaker::processTexture() {
...
         return;
     }
 
-    if (!processedTexture) {
-        handleError("Could not process texture " + _textureURL.toString());
-        return;
-    }
+    // Compressed KTX
+    {
+        // IMPORTANT: _originalTexture is empty past this point
```
Comment (birarda) [inline libraries/baking/src/TextureBaker.cpp]: it looks like this no longer applies here

File: libraries/image/src/image/Image.cpp [hunk 2]
```diff
@@ -284,11 +224,14 @@ QImage processRawImageData(QByteArray&& content, const std::string& filename) {
...
 }
 
-gpu::TexturePointer processImage(QByteArray&& content, const std::string& filename,
+gpu::TexturePointer processImage(std::shared_ptr<QIODevice> content, const std::string& filename,
                                  int maxNumPixels, TextureUsage::Type textureType,
-                                 const std::atomic<bool>& abortProcessing) {
+                                 bool compress, const std::atomic<bool>& abortProcessing) {
 
-    QImage image = processRawImageData(std::move(content), filename);
+    QImage image = processRawImageData(*content.get(), filename);
```
Comment (Atlante45) [inline libraries/image/src/image/Image.cpp]: Not blocking, but the `.get()` is unnecessary here.

File: libraries/image/src/image/Image.cpp [hunk 3]
```diff
@@ -284,11 +224,14 @@ QImage processRawImageData(QByteArray&& content, const std::string& filename) {
...
+    // Texture content can take up a lot of memory. Here we release our ownership of that content
+    // in case it can be released.
+    content.reset();
```
Comment (Atlante45) [inline libraries/image/src/image/Image.cpp]: In the case of the owning buffer, I don't think this will free the memory.

**Comment by huffman** [comment]
I am going to remove uncompressed texture generation for non-cubemaps

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10525/HighFidelity-Beta-PR13283-015c092.apk

**Comment by huffman** [comment]
Please retest!

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10525/HighFidelity-Beta-PR13283-015c092.dmg

**Comment by qlipdahlin** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
Pause Testing

**Comment by qlipdahlin** [comment]
@conklin94122 PASS

Desktop and Android devices loaded the interface and the sandbox. Imported script and desktop saw 4 bottom boxes with success on them while mobile saw 2 top boxes with success on them. Ran through Testrail 998 and passed.

Did encounter an issue with assets stuck in pending on the asset browser for the sandbox, but was found to be related to: https://highfidelity.manuscript.com/f/cases/15842/Sandbox-cannot-be-shut-down-force-quit-only

To finish passing test, used Mac that does not have this issue and loaded the domain for it.

**Comment by birarda** [comment]
@Atlante45 handling review on this

**Pull request merged**